crates/nostr-sdk/src/client/mod.rs

```diff
@@ -1124,8 +1124,8 @@ impl Client {
         let filters: Vec<Filter> = self.get_contact_list_filters().await?;
         let events: Vec<Event> = self.get_events_of(filters, timeout).await?;
 
-        for event in events.into_iter() {
-            for tag in event.into_iter_tags() {
+        if let Some(latest_event) = events.into_iter().max_by_key(|event| event.created_at()) {
```

Comment:
The result of `get_events_of` is always sorted DESC by timestamp (reverse BTreeSet), so here is enough to use `first`. I'll change it during the cherry-picking :)